packages/integrations/src/opnsense/opnsense-integration.ts

```diff
@@ -0,0 +1,189 @@
+    }
+    const summary = opnsenseSystemSummarySchema.parse(await responseSummary.json());
+
+    return {
+      version: summary.versions.at(0) ?? "Unknown",
+    };
+  }
+
+  private getInterfacesChannel() {
+    return createChannelEventHistory<FirewallInterface[]>(`integration:${this.integration.id}:interfaces`, 15);
+  }
+
+  public async getFirewallInterfacesAsync(): Promise<FirewallInterfacesSummary[]> {
+    const channel = this.getInterfacesChannel();
+
+    const interfaceSummary = await fetchWithTrustedCertificatesAsync(this.url("/api/diagnostics/traffic/interface"), {
+      headers: {
+        Authorization: this.getAuthHeaders(),
+      },
+    });
+
+    if (!interfaceSummary.ok) {
+      throw new ResponseError(interfaceSummary);
+    }
+    const test_value = await interfaceSummary.json();
+    const interfaces = opnsenseInterfacesSchema.parse(test_value);
+
+    const returnValue: FirewallInterface[] = [];
+    const interfaceKeys = Object.keys(interfaces.interfaces);
+
+    for (const key of interfaceKeys) {
+      const inter = interfaces.interfaces[key];
+      if (!inter) continue;
+
+      const name = inter.name;
+      const nameValue = typeof name === "string" ? name : "unknown";
+      const bytesTransmitted = inter["bytes transmitted"];
+      const bytesReceived = inter["bytes received"];
+      const receiveValue = typeof bytesReceived === "string" ? parseInt(bytesReceived, 10) : 0;
+      const transmitValue = typeof bytesTransmitted === "string" ? parseInt(bytesTransmitted, 10) : 0;
+
+      returnValue.push({
+        name: nameValue,
+        receive: receiveValue,
+        transmit: transmitValue,
+      });
+    }
+
+    await channel.pushAsync(returnValue);
+
+    return await channel.getSliceAsync(0, 1);
+  }
+
+  public async getFirewallMemoryAsync(): Promise<FirewallMemorySummary> {
+    const responseMemory = await fetchWithTrustedCertificatesAsync(
+      this.url("/api/diagnostics/system/systemResources"),
+      {
+        headers: {
+          Authorization: this.getAuthHeaders(),
+        },
+      },
+    );
+    if (!responseMemory.ok) {
+      throw new ResponseError(responseMemory);
+    }
+
+    const memory = opnsenseMemorySchema.parse(await responseMemory.json());
+
+    // Using parseInt for memoryTotal is normal, the api sends the total memory as a string
+    const memoryTotal = parseInt(memory.memory.total);
+    const memoryUsed = memory.memory.used;
+    const memoryPercent = (100 * memoryUsed) / memoryTotal;
+    return {
+      total: memoryTotal,
+      used: memoryUsed,
+      percent: memoryPercent,
+    };
+  }
+
+  public async getFirewallCpuAsync(): Promise<FirewallCpuSummary> {
+    const cpuSummary = await fetchWithTrustedCertificatesAsync(this.url("/api/diagnostics/cpu_usage/stream"), {
+      headers: {
+        Authorization: this.getAuthHeaders(),
+      },
+    });
+
+    if (!cpuSummary.ok) {
+      throw new Error(`HTTP error! status: ${cpuSummary.status}`);
```

Comment:
```suggestion
      throw new ResponseError(cpuSummary);
```
Please rename cpuSummary to something, where it is obvious that it's a response

Reply:
Done, and I did the same for every response in the same file, with the same naming scheme for every variable.